hazelcast/src/test/java/com/hazelcast/config/YamlConfigBuilderTest.java

```diff
@@ -3953,6 +3996,21 @@ public void testPersistentMemoryConfiguration_SystemMemoryMode() {
         assertEquals(SYSTEM_MEMORY, pmemConfig.getMode());
     }
 
+    @Test(expected = InvalidConfigurationException.class)
+    public void testPersistentMemoryNonExistingValue() {
+        String yaml = ""
+                + "hazelcast:\n"
+                + "  native-memory:\n"
+                + "    persistent-memory:\n"
+                + "      enabled: true\n"
+                + "      mode: SYSTEM_MEMORY1\n";
+
+        Config config = buildConfig(yaml);
+        PersistentMemoryConfig pmemConfig = config.getNativeMemoryConfig().getPersistentMemoryConfig();
+        assertTrue(pmemConfig.isEnabled());
+        assertEquals(SYSTEM_MEMORY, pmemConfig.getMode());
```

Comment:
These are probably unnecessary as the `buildConfig` call would fail

Reply:
fixed